worker/export.go

```diff
@@ -791,6 +772,77 @@ func exportInternal(ctx context.Context, in *pb.ExportRequest, db *badger.DB,
 		})
 	}
 
+	writeSchema := func(prefix byte) error {
+		txn := db.NewTransactionAt(in.ReadTs, false)
+		defer txn.Discard()
+		iopts := badger.DefaultIteratorOptions
+		iopts.Prefix = []byte{prefix}
+		if in.Namespace != math.MaxUint64 {
+			iopts.Prefix = append(iopts.Prefix, x.NamespaceToBytes(in.Namespace)...)
+		}
+
+		itr := txn.NewIterator(iopts)
+		defer itr.Close()
+		for itr.Rewind(); itr.Valid(); itr.Next() {
+			item := itr.Item()
+			// Don't export deleted items.
+			if item.IsDeletedOrExpired() {
+				continue
+			}
+			pk, err := x.Parse(item.Key())
+			if err != nil {
+				glog.Errorf("error %v while parsing key %v during export. Skip.", err,
+					hex.EncodeToString(item.Key()))
+				return err
+			}
+
+			var kvs *bpb.KVList
+			switch prefix {
+			case x.ByteSchema:
+				if !skipZero {
+					if servesTablet, err := groups().ServesTablet(pk.Attr); err != nil || !servesTablet {
+						continue
+					}
+				}
+
+				var update pb.SchemaUpdate
+				err = item.Value(func(val []byte) error {
+					return update.Unmarshal(val)
+				})
+				if err != nil {
+					// Let's not propagate this error. We just log this and continue onwards.
+					glog.Errorf("Unable to unmarshal schema: %+v. Err=%v\n", pk, err)
+					continue
+				}
+				kvs, err = toSchema(pk.Attr, &update)
+
+			case x.ByteType:
+				var update pb.TypeUpdate
+				err := item.Value(func(val []byte) error {
+					return update.Unmarshal(val)
+				})
+				if err != nil {
+					// Let's not propagate this error. We just log this and continue onwards.
+					glog.Errorf("Unable to unmarshal type: %+v. Err=%v\n", pk, err)
+					return nil
+				}
+				kvs, err = toType(pk.Attr, update)
```

Comment:
This error is not being handled.

Reply:
the same for `toSchema` as well

Reply:
They don't return an error. So removed it from the return values of the function.